### PR TITLE
feat-note-editor-done

### PR DESCRIPTION
**Pull request by ankit9015**
Created a basic custom text editor from scratch.
Use console in developers tools to see most of the feature

### CODE REVIEW

File: src/components/LabelButton/LabelButton.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { useState } from "react";
+
+import { MdOutlineLabel } from "../../utils/icons/icons";
+import "./label-button.css";
+
+function LabelButton({ noteData, setNoteData }) {
+  const [newLabel, setNewLabel] = useState("");
+  const [showDropdown, setShowDropdown] = useState(false);
+  console.log(noteData);
```
Comment (munjalSaksham71) [inline src/components/LabelButton/LabelButton.jsx:9]: Remove this log

**Review by munjalSaksham71 — APPROVED** [review]
Everything looks Cool!  Code is also understandable. Just remove the log

File: src/components/LabelButton/LabelButton.jsx (hunk 1 above)
Comment (ankit9015) [reply src/components/LabelButton/LabelButton.jsx:9]: this log is to display the noteData state, will remove it before the merge

File: src/components/Note/Note.jsx [hunk 2]
```diff
@@ -4,34 +4,94 @@ import {
...
 import "./note.css";
 
 function Note() {
+  const defaultState = {
+    title: "",
+    body: "",
+    isPinned: false,
+    tags: ["home", "fruits"],
+    bgColor: "#ffffff",
+    createdAt: new Date().toLocaleDateString(),
```
Comment (dikshadhote) [inline src/components/Note/Note.jsx]: use color name

File: src/components/Note/note.css [hunk 3]
```diff
@@ -1,31 +1,69 @@
...
-  margin: 5px 5px 5px auto;
+  margin: 5px 5px 0 auto;
+}
+
+.note-tags {
+  display: inline-flex;
+  justify-content: space-between;
+  align-items: baseline;
+  border: 2px solid var(--dark-gray);
+  border-radius: 66px;
```
Comment (dikshadhote) [inline src/components/Note/note.css]: use rem for  values greater than 10px

File: src/components/Note/Note.jsx [hunk 4]
```diff
@@ -4,34 +4,94 @@ import {
...
+            }
+            rows="1"
+          />
+          <textarea
+            className="note-body p-xs"
+            type="text"
+            placeholder="Body of the note"
+            value={noteData.body}
+            onChange={(e) => setNoteData({ ...noteData, body: e.target.value })}
+          />
```
Comment (dikshadhote) [inline src/components/Note/Note.jsx]: will suggest to write timer for long text onChange input  as the function will get called less times.

**Review by dikshadhote — CHANGES_REQUESTED** [review]
- Adding label with chips looks cool
- I like that you have given selection panel for card color. But the text color needs to be in contrast with background. If I select black color, then text and icons will not be visible.

If you want to continue with this then you will need mechanism to see if background color is dark or light. Please check feasibility of this feature.

-  Also please decide on colors for primary and seconadary. Login and login as guest can be made as primary. Change text color or background of create new account. they are not going with yellow color
image

File: src/components/Note/Note.jsx (hunk 4 above)
Comment (ankit9015) [reply src/components/Note/Note.jsx]: If we create a timer function to update the value in noteData then nothing will be rendered in that area for that span of time, as the textarea is a controlled input which is reading its value from noteData

**Comment by ankit9015** [comment]
> * Adding label with chips looks cool
> * I like that you have given selection panel for card color. But the text color needs to be in contrast with background. If I select black color, then text and icons will not be visible.
> 
> If you want to continue with this then you will need mechanism to see if background color is dark or light. Please check feasibility of this feature.
> 
> * Also please decide on colors for primary and seconadary. Login and login as guest can be made as primary. Change text color or background of create new account. they are not going with yellow color
>   image

Thanks for the suggestions, I will consider this when I will update the UI after completing the functionality

**Review by dikshadhote — APPROVED** [review]

**Pull request merged**